src/display/display_utils.js

```diff
@@ -65,11 +65,16 @@ class BaseCanvasFactory {
 }
 
 class DOMCanvasFactory extends BaseCanvasFactory {
+  constructor(ownerDocument = globalThis.document) {
+    super();
+    this._document = ownerDocument;
+  }
+
   create(width, height) {
     if (width <= 0 || height <= 0) {
       throw new Error("Invalid canvas size");
     }
-    const canvas = document.createElement("canvas");
+    const canvas = this._document.createElement("canvas");
```

Comment:
Is this actually a *necessary* change for this patch though, since I cannot imagine that the exact `document` which you use to just *create* the element matters!?

This would e.g. make the `DOMCanvasFactory` and `NodeCanvasFactory` interfaces slightly different (which you'd want to avoid).

Reply:
Yes. The canvas only has access to the resources owned by the creating (owner) document. So, calls to measure/draw text will be inaccurate without this. 

I didn't just change this without cause, I still had missing fonts on some pages. Patching the creation of canvases to use the owner document fixes it.

The interface is the same. They have an optional argument for the constructor (Node version doesn't care, so it ignores) I could explicitly add that to the node constructor?

Reply:
If you insert (adopt) a canvas created in a foreign document into a new document, it will gain access to fonts loaded in your document and lose access to fonts from its original document. This point is especially important for scratch/off-screen canvases.